Feature/is34 jpa entity modeling

KiWooyoung opened the pull request:
## Description
* 엔티티 맵핑
* 코스 리스트 API - GET

## Related Tickets
* 예약을 위한 코스 리스트 조회 API 구현하기 (#26)
* JPA Entity Modeling하기 (#34)

<br>

> #### Reference
> * [JPA / Hibernate One to One Mapping Example with Spring Boot](https://www.callicoder.com/hibernate-spring-boot-jpa-one-to-one-mapping-example)

---

<br>

## Check List
- [x] PR 제목은 유의미한가?
- [ ] PR 내용만 보고도 이해할 수 있을 정도로 기술되었는가?
- [x] 관련 issue를 적었는가?
- [x] 관련 reference가 있다면 함께 적었는가?
- [x] Reviewer, Assginess, Label을 지정했는가?

---

KiWooyoung committed:
[#34][ADD] tourist 테이블 맵핑

KiWooyoung committed:
[#34][ADD] price 테이블 맵핑

KiWooyoung committed:
[#34][ADD] course 테이블 맵핑

KiWooyoung committed:
[#34][ADD] company 테이블 맵핑

KiWooyoung committed:
[#34][ADD] booking 테이블 맵핑

KiWooyoung committed:
[#34][ADD] course_image 테이블 맵핑

KiWooyoung committed:
[#34][ADD] BaseTime 추가

KiWooyoung committed:
[#26][ADD] 코스 List 조회 API

KiWooyoung committed:
[34][ADD] DTO 추가

KiWooyoung committed:
[][CLN] common, config 경로 나눔 / 기타 클래스 생성

KiWooyoung committed:
[#34][ENH] 수정

KiWooyoung committed:
[#34][ADD] station 맵핑

KiWooyoung committed:
[#34][ADD] station_image 맵핑

KiWooyoung committed:
[#34][ADD] info_type 맵핑

KiWooyoung committed:
[#34][ADD] 맵핑 완료 -

일단 맵핑은 다 했습니다.
추가 적으로 수정해야할 부분 많은거 같은데, 앞으로 해볼게요.
그리고,
프로퍼티에 'JPA' 부분을 주석해야지 실행 되더라고요. 일단 주석 처리 햇는데,
추가적으로 알아 가볼께요.

opklnm102 commented on build.gradle:
```diff
@@ -26,20 +26,13 @@ repositories {
     compile('org.springframework.boot:spring-boot-starter-web')
-
-    // persitence
+    compile('org.springframework.boot:spring-boot-starter-jdbc')
```
이건 어디서 사용해요?

KiWooyoung replied:
아 프로퍼티 때문에 안되길래 저거 없어서 그랬나 하고 넣어봤어요

opklnm102 commented on src/main/java/kr/co/mashup/mapc/controller/BookingController.java:
```diff
@@ -0,0 +1,26 @@
+
+    private BookingService bookingService;
+
+    @GetMapping(path = "/course/list")
```
/booking/course 는 어떤가요?

KiWooyoung replied:
저도 둘중에 고민했는데, 코스의 list 를 가져온다는 생각으로 list로 했어요ㅋㅋㅋ

opklnm102 commented on src/main/java/kr/co/mashup/mapc/controller/BookingController.java:
```diff
@@ -0,0 +1,26 @@
+
+@RestController
+@AllArgsConstructor
+@Slf4j
```
이건 사용처가 없는듯..?

opklnm102 commented on src/main/java/kr/co/mashup/mapc/dto/CourseDto.java:
```diff
@@ -0,0 +1,39 @@
+    @Data
+    @ToString
+    @Slf4j
+    public static class CourseSelection {
```
static inner class로 한 이유가 있나요?

KiWooyoung replied:
코스 관련된 여러 액션이 있을거 같아서, 모으면 어떨까 하고 이너 클래스로 해봤어욥

opklnm102 commented on src/main/java/kr/co/mashup/mapc/dto/CourseDto.java:
```diff
@@ -0,0 +1,39 @@
+
+    @Data
+    @ToString
+    @Slf4j
```
요것도 불필요한듯..

KiWooyoung replied:
아하 얍스. 저거 로그 찍어보다가 ㅋㅋㅋ

opklnm102 commented on src/main/resources/application.properties:
```diff
@@ -3,10 +3,11 @@
-spring.datasource.username=mapc_db
-spring.datasource.password=mapc_db1234
+spring.datasource.url=jdbc:mysql://key-rds.cy2pmafzldzf.ap-northeast-2.rds.amazonaws.com/SeoulAppContest?useUnicode=true&charaterEncoding=utf-8&useSSL=false
+spring.datasource.username=keywy
```
ㅋㅋㅋㅋ.... 우리 RDS 비번 user name, pw 바꿔야할듯요

KiWooyoung replied:
ㅋㅋㅋㅋ넵

opklnm102 replied:
DB 해킹되겠다
전세계적으로 노출되...

TODO 패스워드 아디디 공개 안할 수 없나~? -> 이건 제가 작업해놓을께요

opklnm102 replied:
TODO 패스워드 아디디 공개 안할 수 없나~? -> 이건 제가 작업해놓을께요

opklnm102 commented on src/main/resources/application.properties:
```diff
@@ -18,16 +19,16 @@ spring.datasource.hikari.idle-timeout=300000
+#spring.jpa.show-sql=true
+#spring.jpa.properties.hibernate.format_sql=true
+#spring.jpa.open-in-view=false
+#spring.data.jpa.repositories.enabled= true
```
해당 properties들 모두 주석처리 되어 있는데...
Entity들 제대로 동작하는건가요..?

KiWooyoung replied:
마지막 커밋메시지에 dae68b7 아래 내용에 이유를 적어놓긴했는데. 이부분 탐구해야 될 부분입니다.

opklnm102 replied:
PR에서 commit message가 쉽게 보이지가 않아서.... 나중에 봤어요.
이런 메시지는 commit message보다는 PR body에 들어가 있었으면 훨씬 좋았을듯?

저거 있으면 어떤 에러 메시지가 나오는지 첨부해줄 수 있나요?

opklnm102 replied:
저거 있으면 어떤 에러 메시지가 나오는지 첨부해줄 수 있나요?

KiWooyoung replied:
주석 처리해야지 실행이 되서 주석하긴했는데, 저도 이부분 알아봐야겠어요. 또 프로퍼티 설정 같은것들도

KiWooyoung replied:
아하 오케 알겠습니당

opklnm102 replied:
```
	at kr.co.mashup.mapc.MapcApplication.main(MapcApplication.java:14) [classes/:na]
Caused by: javax.persistence.PersistenceException: [PersistenceUnit: default] Unable to build Hibernate SessionFactory; nested exception is org.hibernate.tool.schema.spi.SchemaManagementException: Schema-validation: wrong column type encountered in column [id] in table [booking]; found [int (Types#INTEGER)], but expecting [bigint (Types#BIGINT)]
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.buildNativeEntityManagerFactory(AbstractEntityManagerFactoryBean.java:402) ~[spring-orm-5.0.8.RELEASE.jar:5.0.8.RELEASE]
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.afterPropertiesSet(AbstractEntityManagerFactoryBean.java:377) ~[spring-orm-5.0.8.RELEASE.jar:5.0.8.RELEASE]
	at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.afterPropertiesSet(LocalContainerEntityManagerFactoryBean.java:341) ~[spring-orm-5.0.8.RELEASE.jar:5.0.8.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1758) ~[spring-beans-5.0.8.RELEASE.jar:5.0.8.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1695) ~[spring-beans-5.0.8.RELEASE.jar:5.0.8.RELEASE]
	... 16 common frames omitted
Caused by: org.hibernate.tool.schema.spi.SchemaManagementException: Schema-validation: wrong column type encountered in column [id] in table [booking]; found [int (Types#INTEGER)], but expecting [bigint (Types#BIGINT)]
	at org.hibernate.tool.schema.internal.AbstractSchemaValidator.validateColumnType(AbstractSchemaValidator.java:159) ~[hibernate-core-5.2.17.Final.jar:5.2.17.Final]
	at org.hibernate.tool.schema.internal.AbstractSchemaValidator.validateTable(AbstractSchemaValidator.java:143) ~[hibernate-core-5.2.17.Final.jar:5.2.17.Final]
	at org.hibernate.tool.schema.internal.GroupedSchemaValidatorImpl.validateTables(GroupedSchemaValidatorImpl.java:42) ~[hibernate-core-5.2.17.Final.jar:5.2.17.Final]
	at org.hibernate.tool.schema.internal.AbstractSchemaValidator.performValidation(AbstractSchemaValidator.java:89) ~[hibernate-core-5.2.17.Final.jar:5.2.17.Final]
	at org.hibernate.tool.schema.internal.AbstractSchemaValidator.doValidation(AbstractSchemaValidator.java:68) ~[hibernate-core-5.2.17.Final.jar:5.2.17.Final]
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:191) ~[hibernate-core-5.2.17.Final.jar:5.2.17.Final]
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:72) ~[hibernate-core-5.2.17.Final.jar:5.2.17.Final]
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:312) ~[hibernate-core-5.2.17.Final.jar:5.2.17.Final]
	at org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:462) ~[hibernate-core-5.2.17.Final.jar:5.2.17.Final]
	at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:892) ~[hibernate-core-5.2.17.Final.jar:5.2.17.Final]
	at org.springframework.orm.jpa.vendor.SpringHibernateJpaPersistenceProvider.createContainerEntityManagerFactory(SpringHibernateJpaPersistenceProvider.java:57) ~[spring-orm-5.0.8.RELEASE.jar:5.0.8.RELEASE]
	at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.createNativeEntityManagerFactory(LocalContainerEntityManagerFactoryBean.java:365) ~[spring-orm-5.0.8.RELEASE.jar:5.0.8.RELEASE]
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.buildNativeEntityManagerFactory(AbstractEntityManagerFactoryBean.java:390) ~[spring-orm-5.0.8.RELEASE.jar:5.0.8.RELEASE]
	... 20 common frames omitted
```

opklnm102 replied:
안돼는 이유가 이거 같은데... booking table의 id column type이 int(11) 이잖아요?
```java
@Entity
public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    ...
```
근데 Booking Entity에 Long으로 되어 있어서 에러가 나는거고 
Long은 BIGINT에 매핑되요

KiWooyoung replied:
아하 그렇군요 Long : BIGINT / Integer : INT 인가 보네요. 좀 더 자료 찾아보고 한번 정리해볼게요.

opklnm102 commented on src/main/java/kr/co/mashup/mapc/repository/BookingRepository.java:
```diff
@@ -0,0 +1,9 @@
+
+public interface BookingRepository extends JpaRepository<Course, Long> {
+
+    //TODO 필요한 컬럼만 가져올려면? 찾아보자 -- findALl 한다음 데이터 DTO 에 담는게 불필요 한거 아닌가?
```
좋은 접근인듯..!
방법 찾으면 공유해주세요

opklnm102 commented on src/main/java/kr/co/mashup/mapc/dto/DataResponse.java:
```diff
@@ -0,0 +1,11 @@
+@Data //TODO @Data 탐구
+public class DataResponse<T> {
+    private T Result;
+    private List<T> Results;
```
?!

KiWooyoung replied:
base response 를 만들어 보면 어떨까 하고 만들었고,, 음.. 그 @Data는 제가 롬복을 잘 안써봐서 한번 제대로 알고 가자? 라는 식으로 todo 해놨네요..

opklnm102 replied:
아.. 변수명 때문에...   ?! 달아놨어요

opklnm102 reviewed (commented):
아하 고생하셨어요. 해당 PR은 내일 천천히 이야기해봐요

opklnm102 commented:
우리 해당 PR로 내일 페어코딩 잠깐 해봐요

KiWooyoung commented:
맨날 혼자 코딩하는데 이렇게 리뷰 해주니까 쩌는듯요 ㅋㅋㅋ

Commit:
[#34][BUG] 에러 - No identifier specified for entity (#44)

식별 키 (PK) 가 없는 에러 같아서 테이블에 id를 추가하고, 맵핑하니 에러는 안난다..
근데 PK가 꼭 있어야 맵핑되나..? 더 찾아볼수도 있지만 밤이라서 이대로 fix..

opklnm102 committed:
Merge branch 'develop' into feature/is34-jpa-entity-modeling

# Conflicts:
#	src/main/resources/application.properties

opklnm102 committed:
[#34][ADD] Add JpaAuditing configuration

opklnm102 committed:
[#34][REFACTOR] entity 리펙토링

opklnm102 committed:
[#34][ADD] Add repository test case

opklnm102 committed:
[#34][FIx] entity의 auot increment field id로 네이밍 통일되도록 수정

opklnm102 committed:
Merge branch 'develop' into feature/is34-jpa-entity-modeling

# Conflicts:
#	src/main/java/kr/co/mashup/mapc/dto/CourseDto.java
#	src/main/java/kr/co/mashup/mapc/entity/BaseTimeEntity.java
#	src/main/java/kr/co/mashup/mapc/entity/Booking.java
#	src/main/java/kr/co/mashup/mapc/entity/Company.java
#	src/main/java/kr/co/mashup/mapc/entity/Course.java
#	src/main/java/kr/co/mashup/mapc/entity/CourseImage.java
#	src/main/java/kr/co/mashup/mapc/entity/Information.java
#	src/main/java/kr/co/mashup/mapc/entity/Price.java
#	src/main/java/kr/co/mashup/mapc/entity/Station.java
#	src/main/java/kr/co/mashup/mapc/entity/StationImage.java
#	src/main/java/kr/co/mashup/mapc/entity/Tourist.java
#	src/main/java/kr/co/mashup/mapc/repository/BookingRepository.java
#	src/main/java/kr/co/mashup/mapc/repository/TouristRepository.java
#	src/main/java/kr/co/mashup/mapc/service/BookingService.java

opklnm102 committed:
[#34][FIX] develop merge시 생성된 불필요 파일 제거 및 모든 TC 성공하도록 수정
